unitTests/ui/MissingTermsTest.ts

```diff
@@ -0,0 +1,146 @@
+import * as Mock from '../MockEnvironment';
+import { MissingTerms } from '../../src/ui/MissingTerm/MissingTerms';
+import { $$ } from '../../src/Core';
+import { FakeResults } from '../Fake';
+
+export function MissingTermsTest() {
+  describe('MissingTerm', () => {
+    let test: Mock.IBasicComponentSetup<MissingTerms>;
+    let fakeResult = FakeResults.createFakeResult();
+    fakeResult.state.q = 'This will be a really really really long query';
+    const wordNotInQuery = 'foo';
```

Comment:
It would be important to test much more corner case here to ensure the quoting mechanism works correctly.

* Repeated terms
* Terms "enclosed" inside another term (eg: we have keywords `the` and `theory` in the search box, if somebody wants to force `the` to appear, if should not quote `"the"ory`
* Other languages (chinese)
* Terms with hyphens or other special characters.

Try to break your code as much as possible :)